skywalking/plugins/sw_fastapi.py

```diff
@@ -0,0 +1,73 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from starlette.types import ASGIApp, Receive, Scope, Send
+from skywalking import Layer, Component, config
+from skywalking.trace.carrier import Carrier
+from skywalking.trace.context import get_context, NoopContext
+from skywalking.trace.span import NoopSpan
+from skywalking.trace.tags import TagHttpMethod, TagHttpURL, TagHttpStatusCode, TagHttpParams
+
+link_vector = ['https://flask.palletsprojects.com']
+support_matrix = {
+    'fastapi': {
+        '>=3.6': ['1.1', '2.0']  # 1.1 to be removed in near future
+    }
+}
+note = """"""
+
+
+def install():
+    from fastapi import routing
+    from fastapi.routing import APIRoute
+
+    _handle = APIRoute.handle
+
+    def params_tostring(params):
+        return '\n'.join([f"{k}=[{','.join(params.getlist(k))}]" for k, _ in params.items()])
+
+    async def _sw_handle(self, scope: Scope, receive: Receive, send: Send):
+        from starlette.requests import Request
+        req = Request(scope, receive=receive, send=send)
+        carrier = Carrier()
+        method = req.method
+
+        for item in carrier:
+            if item.key.capitalize() in req.headers:
+                item.val = req.headers[item.key.capitalize()]
+
+        span = NoopSpan(NoopContext()) if config.ignore_http_method_check(method) \
+            else get_context().new_entry_span(op=dict(scope)["path"], carrier=carrier, inherit=Component.General)
+
+        with span:
+            span.layer = Layer.Http
+            span.component = Component.Flask
+            span.peer = f"{req.client.host}:{req.client.port}"
+            span.tag(TagHttpMethod(method))
+            span.tag(TagHttpURL(req.url._url.split('?')[0]))
+            if config.fastapi_collect_http_params and req.values:
```

Comment:
Where is `fastapi_collect_http_params` declared? I can't find on in the config.

Reply:
Also, https://github.com/apache/skywalking-python/blob/master/docs/en/setup/EnvVars.md should be updated accordingly.

Reply:
request_response has not the request args ，this function need function as args,  inner app has the request args, but is the app args can use？

Reply:
I just mean the parameter of `fastapi_collect_http_params` for the agent user. 
I am not a qualified reviewer for the Python agent, so, I could be wrong. Mostly, I am just chasing for a good document.

Reply:
> request_response has not the request args ，this function need function as args,  inner app has the request args, but is the app args can use？

I think you are replying the other question. When you want to reply please make sure to reply the right context. Go to the message that you want to reply and click the menu ... in the top-right and choose "quote and reply". Otherwise people may get very confused when reading your comments.